storm-core/src/clj/org/apache/storm/daemon/logviewer.clj

```diff
@@ -502,6 +503,20 @@
       (-> (resp/response "Page not found")
           (resp/status 404)))))
 
+(defnk set-log-file-permissions [fname root-dir]
+       (let [file (.getCanonicalFile (File. root-dir fname))
+             run-as-user (*STORM-CONF* SUPERVISOR-RUN-WORKER-AS-USER)
+             parent (.getParent (File. root-dir fname))
+             md-file (if (nil? parent) nil (get-metadata-file-for-wroker-logdir parent))
+             topo-owner (if (nil? md-file) nil (get-topo-owner-from-metadata-file (.getCanonicalPath md-file)))]
+         (if (and run-as-user
```

Comment:
Nit: since there is no 'else' branch, replacing `if` with `when` will allow you to remove the `do`.